tasks/WaypointNavigator.cpp

```diff
@@ -71,8 +71,8 @@ void WaypointNavigator::updateHook()
                 waypoints.pop_front();
                 if (waypoints.size() == 0){
                     keep_position = true;
-                    if(state() != KEEP_WAYPOINT){
-                        state(KEEP_WAYPOINT);
+                    if(state() == KEEP_WAYPOINT){
```

Comment:
This changes the behavior of the task a bit, allowing it only to go to finished when it was in KEEP_WAYPOINT before.
Not sure if this constraint is necessary.